scripts/ci/sync_extensions.py

```diff
@@ -0,0 +1,146 @@
+        if search_result is not None:
+            url = search_result.group(1)
+            extension_name = re.findall(NAME_REGEX, url)[0].replace('_', '-')
+            updated_exts.append(extension_name)
+    return updated_exts
+
+
+def _download_file(url, file_path):
+    import requests
+    count = 3
+    the_ex = None
+    while count > 0:
+        try:
+            response = requests.get(url, stream=True, allow_redirects=True)
+            assert response.status_code == 200, "Response code {}".format(response.status_code)
+            break
+        except Exception as ex:
+            the_ex = ex
+            count -= 1
+    if count == 0:
+        raise Exception("Request for {} failed: {}".format(url, str(the_ex)))
+
+    with open(file_path, 'wb') as f:
+        for chunk in response.iter_content(chunk_size=1024):
+            if chunk:  # ignore keep-alive new chunks
+                f.write(chunk)
+
+
+def _sync_wheel(ext, updated_indexes, client, overwrite, temp_dir):
+    download_url = ext['downloadUrl']
+    whl_file = download_url.split('/')[-1]
+    whl_path = os.path.join(temp_dir, whl_file)
+    _download_file(download_url, whl_path)
+    if not overwrite:
+        exists = client.exists(container_name=STORAGE_CONTAINER, blob_name=whl_file)
+        if exists:
+            print("Skipping '{}' as it already exists...".format(whl_file))
+            return
+    client.create_blob_from_path(container_name=STORAGE_CONTAINER, blob_name=whl_file,
+                                 file_path=os.path.abspath(whl_path))
+    url = client.make_blob_url(container_name=STORAGE_CONTAINER, blob_name=whl_file)
+    updated_index = ext
+    updated_index['downloadUrl'] = url
+    updated_indexes.append(updated_index)
+
+
+def _update_target_extension_index(updated_indexes, target_index_path):
+    NAME_REGEX = r'.*/([^/]*)-\d+.\d+.\d+'
+    with open(target_index_path, 'r') as infile:
+        curr_index = json.loads(infile.read())
+    for entry in updated_indexes:
+        url = entry['downloadUrl']
+        extension_name = re.findall(NAME_REGEX, url)[0].replace('_', '-')
+        if extension_name not in curr_index['extensions'].keys():
+            print("Adding '{}' to index...".format(extension_name))
+            curr_index['extensions'][extension_name] = [entry]
+        else:
+            print("Updating '{}' in index...".format(extension_name))
+            if curr_index['extensions'][extension_name][-1]['filename'] == entry['filename']:  # in case of overwrite
+                curr_index['extensions'][extension_name][-1] = entry
+            else:
+                curr_index['extensions'][extension_name].append(entry)
+
+    with open(os.path.join(target_index_path), 'w') as outfile:
+        outfile.write(json.dumps(curr_index, indent=4, sort_keys=True))
+
+
+def main():
+    import shutil
+    import tempfile
+    from azure.storage.blob import BlockBlobService
```

Comment:
install azure.storage.blob in Bash task?

Reply:
yes, as a prerequisite.